cmd/snap-update-ns/common.go

```diff
@@ -0,0 +1,78 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/osutil"
+)
+
+type CommonProfileUpdate struct {
+	// instanceName is the name of the snap or instance to update.
+	instanceName string
+
+	currentProfilePath string
+	desiredProfilePath string
+}
+
+func (up *CommonProfileUpdate) Lock() (unlock func(), err error) {
+	return func() {}, nil
+}
+
+func (up *CommonProfileUpdate) Assumptions() *Assumptions {
+	return nil
+}
+
+// NeededChanges computes the sequence of mount changes needed to transform current profile to desired profile.
+func (up *CommonProfileUpdate) NeededChanges(current, desired *osutil.MountProfile) []*Change {
+	return NeededChanges(current, desired)
```

Comment:
will this implementation vary between Systen and User in later changes?

Reply:
Initially no but I suspect it will have to as soon as we will begin the Nvidia transition. We need to be able to migrate from the world where Nvidia was a bind mount / symlink farm in one version of snapd to the content-based version that is represented in the mount profile.

Though with all honesty this is here to completely encapsulate the consumer of the interface. It makes testing easy.

Reply:
Do I understand correctly that is purely functional and doesn't read the disk in itself?  I understand the slight convenience for tests but it seems clearer if it was not part of this.  It's not using/referring to `up` at all.

Reply:
It's purely functional, yes.

I can drop it but it would be much more convenient to go through the split branch and then apply changes.

Reply:
You mean land this one and do a follow up? or you mean go over the full chain and then undo things? the latter sounds slightly messy to mess approach ways.

Reply:
I would prefer to land the whole split branch and then remove this from the interface.

Reply:
Updated now.